File: app/src/main/java/ru/prsolution/winstrike/mvp/presenters/RegisterPresenter.java
package ru.prsolution.winstrike.mvp.presenters;

import ru.prsolution.winstrike.mvp.models.LoginModel;
import ru.prsolution.winstrike.mvp.models.MessageResponse;
import ru.prsolution.winstrike.mvp.apimodels.AuthResponse;
import ru.prsolution.winstrike.mvp.apimodels.ConfirmSmsModel;
import ru.prsolution.winstrike.mvp.views.RegisterView;
import ru.prsolution.winstrike.networking.NetworkError;
import ru.prsolution.winstrike.networking.Service;
import rx.Subscription;
import rx.subscriptions.CompositeSubscription;

public class RegisterPresenter {
    private final Service service;
    private final RegisterView view;
    private CompositeSubscription subscriptions;

    public RegisterPresenter(Service service, RegisterView view) {
        this.service = service;
        this.view = view;
        this.subscriptions = new CompositeSubscription();
    }

    public void createUser(LoginModel user) {

        Subscription subscription = service.createUser(new Service.RegisterCallback() {
            @Override
            public void onSuccess(AuthResponse authResponse) {
                view.onRegisterSuccess(authResponse);
            }

            @Override
            public void onError(NetworkError networkError) {
                view.onRegisterFailure(networkError.getAppErrorMessage());
            }

        }, user);

        subscriptions.add(subscription);
    }


    /**
     *  Send code by tap on "Again button send"
     * @param smsModel
     */
    public void sendSms(ConfirmSmsModel smsModel) {

        Subscription subscription = service.sendSmsByUserRequest(new Service.SmsCallback() {
            @Override
            public void onSuccess(MessageResponse authResponse) {
                view.onSendSmsSuccess(authResponse);
            }

            @Override
            public void onError(NetworkError networkError) {
                view.onSmsSendFailure(networkError.getAppErrorMessage());
            }

        }, smsModel);

        subscriptions.add(subscription);
    }



    public void onStop() {
        subscriptions.unsubscribe();
    }

}

File: app/src/main/kotlin/ru/prsolution/winstrike/presentation/login/RegisterPresenter.kt
package ru.prsolution.winstrike.presentation.login

import ru.prsolution.winstrike.datasource.model.login.ConfirmSmsModel
import ru.prsolution.winstrike.domain.models.login.LoginModel
import rx.subscriptions.CompositeSubscription

class RegisterPresenter() {
	private val subscriptions: CompositeSubscription

	init {
		this.subscriptions = CompositeSubscription()
	}

	fun createUser(user: LoginModel) {

/*		val subscription = service.createUser(object : Service.RegisterCallback {
			override fun onSuccess(authResponse: AuthResponse) {
				view.onRegisterSuccess(authResponse)
			}

			override fun onError(networkError: NetworkError) {
				view.onRegisterFailure(networkError.appErrorMessage)
			}

		}, user)

		subscriptions.add(subscription)*/
	}


	/**
	 * Send code by tap on "Again button send"
	 * @param smsModel
	 */
	fun sendSms(smsModel: ConfirmSmsModel) {

/*		val subscription = service?.sendSmsByUserRequest(object : Service.SmsCallback {
			override fun onSuccess(authResponse: MessageResponse) {
//				view.onSendSmsSuccess(authResponse)
			}

			override fun onError(networkError: NetworkError) {
//				view.onSmsSendFailure(networkError.appErrorMessage)
			}

		}, smsModel)

		subscriptions.add(subscription)*/
	}


	fun onStop() {
		subscriptions.unsubscribe()
	}

}
